core/src/storage2/lazy/entry.rs

```diff
@@ -0,0 +1,277 @@
+    pub fn is_mutated(self) -> bool {
+        match self {
+            EntryState::Mutated => true,
+            EntryState::Preserved => false,
+        }
+    }
+
+    /// Returns `true` if the entry state is preserved.
+    pub fn is_preserved(self) -> bool {
+        !self.is_mutated()
+    }
+}
+
+impl<T> SpreadLayout for Entry<T>
+where
+    T: SpreadLayout,
+{
+    const FOOTPRINT: u64 = <T as SpreadLayout>::FOOTPRINT;
+
+    fn pull_spread(ptr: &mut KeyPtr) -> Self {
+        let root_key = ptr.next_for::<Self>();
+        Self::new(pull_spread_root_opt::<T>(&root_key), EntryState::Preserved)
+    }
+
+    fn push_spread(&self, ptr: &mut KeyPtr) {
+        let old_state = self.replace_state(EntryState::Preserved);
+        if old_state.is_mutated() {
+            let root_key = ptr.next_for::<Self>();
+            push_spread_root_opt::<T>(self.value().into(), &root_key);
+        }
+    }
+
+    fn clear_spread(&self, ptr: &mut KeyPtr) {
+        let root_key = ptr.next_for::<Self>();
+        clear_spread_root_opt::<T, _>(&root_key, || self.value().into());
+    }
+}
+
+impl<T> scale::Encode for Entry<T>
+where
+    T: scale::Encode,
+{
+    #[inline]
+    fn size_hint(&self) -> usize {
+        <Option<T> as scale::Encode>::size_hint(&self.value)
+    }
+
+    #[inline]
+    fn encode_to<O: scale::Output>(&self, dest: &mut O) {
+        <Option<T> as scale::Encode>::encode_to(&self.value, dest)
+    }
+
+    #[inline]
+    fn encode(&self) -> Vec<u8> {
+        <Option<T> as scale::Encode>::encode(&self.value)
+    }
+
+    #[inline]
+    fn using_encoded<R, F: FnOnce(&[u8]) -> R>(&self, f: F) -> R {
+        <Option<T> as scale::Encode>::using_encoded(&self.value, f)
+    }
+}
+
+impl<T> scale::Decode for Entry<T>
+where
+    T: scale::Decode,
+{
+    fn decode<I: scale::Input>(input: &mut I) -> Result<Self, scale::Error> {
+        Ok(Self::new(
+            <Option<T> as scale::Decode>::decode(input)?,
+            EntryState::Preserved,
+        ))
+    }
+}
+
+impl<T> PackedLayout for Entry<T>
+where
+    T: PackedLayout,
+{
+    #[inline]
+    fn pull_packed(&mut self, at: &Key) {
+        PackedLayout::pull_packed(&mut self.value, at)
+    }
+
+    #[inline]
+    fn push_packed(&self, at: &Key) {
+        PackedLayout::push_packed(&self.value, at)
+    }
+
+    #[inline]
+    fn clear_packed(&self, at: &Key) {
+        PackedLayout::clear_packed(&self.value, at)
+    }
+}
+
+impl<T> Entry<T>
+where
+    T: PackedLayout,
+{
+    /// Pulls the entity from the underlying associated storage as packed representation.
+    ///
+    /// # Note
+    ///
+    /// Mainly used by lazy storage abstractions that only allow operating on
+    /// packed storage entities such as [`LazyIndexMap`] or [`LazyArray`].
+    pub fn pull_packed_root(root_key: &Key) -> Self {
+        let mut entry =
+            Self::new(pull_packed_root_opt::<T>(root_key), EntryState::Preserved);
+        PackedLayout::pull_packed(&mut entry, root_key);
```

Comment:
I think this is redundant, since the inner `T` will have `pull_packed` called via `pull_packed_root_opt` if it has a value. Certainly the other two methods don't call their respective methods. This would make the `impl<T> PackedLayout for Entry<T>` redundant too but might want to keep it for completeness.